### DISCUSSION TITLE
Tabpanes not rendering with text in markdown

### DISCUSSION THREAD

**Discussion by jonagard** [google/docsy#1135] [Q&A] [answered]
Hi.  I'm trying to use the tabpanes shortcode to render markdown.  Instead of seeing rendered markdown, I am seeing a box with html in it:

<img width="1034" alt="Screen Shot 2022-07-28 at 1 49 13 PM" src="https://user-images.githubusercontent.com/20245240/181603977-4e59a004-f6e3-44ec-ba2f-5108a9177c01.png">

I am just starting with this example which is clearly working on this site:

https://fluxcd.io/docs/installation/#install-the-flux-cli

<img width="996" alt="Screen Shot 2022-07-28 at 1 51 18 PM" src="https://user-images.githubusercontent.com/20245240/181604386-e49a7943-e9d1-4a5c-a4f8-e326b15ab12b.png">

I have pasted that code as-is into the [docsy-example](https://github.com/google/docsy-example) and [SeleniumHQ](https://github.com/SeleniumHQ/seleniumhq.github.io) sites.   On both, I see the html.  Here is the SeleniumHQ one:

<img width="1202" alt="Screen Shot 2022-07-28 at 1 51 42 PM" src="https://user-images.githubusercontent.com/20245240/181604474-094a6565-2a63-4333-be83-b0f72f030bc7.png">

I'm on Hugo v0.101.0+extended:

```
> hugo serve
Start building sites … 
hugo v0.101.0+extended darwin/amd64 BuildDate=unknown
...
```

With Docsy v0.4.0:

```
docsy-example> grep docsy go.mod
module github.com/google/docsy-example
require github.com/google/docsy v0.4.0 // indirect
```

I can get the tabpane to render correctly if I put it in a file at https://github.com/google/docsy/userguide.

I've read the tabpane instructions, and I don't see anything specified that must be included in the config.toml.  I've gone through the config.toml files for docsy and docsy-example.  I found all differences and tried modding the file in docsy-example but nothing makes it work.  I've compared dependencies as best as I can and even copied the versions from docsy to docsy-example, restarted the server after a new `npm install` and it still doesn't render correctly.

Does anyone know what might fix the tabpane so it renders markdown as expected?

**Comment by jonagard** [comment]
FWIW, I read through discussion #1124 out of curiosity.  I setup a brand new site from scratch using that method and copied the tabpane code to _index.md.  This renders as expected.

<img width="1478" alt="Screen Shot 2022-07-28 at 3 02 14 PM" src="https://user-images.githubusercontent.com/20245240/181617127-7a657d04-608f-41a2-8fed-42bd2f2c92ae.png">
 
Then I setup a site from scratch using the instructions at https://www.docsy.dev/docs/get-started/docsy-as-module/start-from-scratch/#tldr-setup-for-the-impatient-expert

I put the same file in and it renders incorrectly:

<img width="1517" alt="Screen Shot 2022-07-28 at 3 04 11 PM" src="https://user-images.githubusercontent.com/20245240/181617476-b8326317-7c7f-4645-90d5-765c2cbdf459.png">

**Reply by deining** [reply]
To me this looks like an issue with hugo's module cache. Support for html text inside tabs was introduced only recently, and maybe you are still using an outdated shortcode version served from the module cache?
You may try to clean the module cache by issueing this command:

```
hugo mod clean
```

Does this cure the problem?

**Note:** This is just a guess, I am not at my desk right now.

**Reply by jonagard** [reply]
Thanks for the reply!

Unfortunately, it didn't seem to fix it.

```
> hugo mod clean
hugo: cleaned module cache for "github.com/FortAwesome/Font-Awesome"
hugo: cleaned module cache for "github.com/google/docsy"
hugo: cleaned module cache for "github.com/google/docsy/dependencies"
hugo: cleaned module cache for "github.com/twbs/bootstrap"
> hugo serve    
hugo: collected modules in 1411 ms
Start building sites … 
hugo v0.101.0+extended darwin/amd64 BuildDate=unknown

                   | FA | NO | EN  
-------------------+----+----+-----
  Pages            | 28 | 94 | 75  
  Paginator pages  |  0 |  0 |  0  
  Non-page files   |  3 |  1 |  3  
  Static files     | 40 | 40 | 40  
  Processed images |  9 |  2 |  9  
  Aliases          |  3 |  0 |  3  
  Sitemaps         |  2 |  1 |  1  
  Cleaned          |  0 |  0 |  0  

Built in 671 ms
Watching for changes in /Users/jonagard/docsy-example/{assets,content,layouts,package.json}
Watching for config changes in /Users/jonagard/docsy-example/config.toml, /Users/jonagard/docsy-example/go.mod
Environment: "development"
Serving pages from memory
Running in Fast Render Mode. For full rebuilds on change: hugo server --disableFastRender
Web Server is available at //localhost:1313/ (bind address 127.0.0.1)
Press Ctrl+C to stop
```

And I still see this when the page loads:

<img width="1018" alt="Screen Shot 2022-07-29 at 3 45 34 PM" src="https://user-images.githubusercontent.com/20245240/181833031-9b750171-3b32-4f19-8aa4-3e312951c88b.png">

**Reply by JosephKav** [reply]
Did you ever get this fixed? I'm keen to get these tabpanes working but mine are rendering just like yours 😞

**Reply by jonagard** [reply]
No :-(   I couldn't get them working.  I've moved forward with the idea that tabpanes just aren't available and I'll have to lay things out without them.

**Comment by deining** [comment, marked as answer]
I could reproduce the problem and found the reason: version `v0.4.0` does not contain the latest changes to the tabpane shortcodes. All you have to do to get it working is update your module to at least the revision `1b10f6bebe`:

```
hugo mod get -u github.com/google/docsy@1b10f6bebe
```

Afterwards, rendering of content in markdown should work as expected.

**Reply by jonagard** [reply]
Thank you!  Yes, this fixes the issue:

<img width="1034" alt="Screen Shot 2022-09-12 at 10 47 54 AM" src="https://user-images.githubusercontent.com/20245240/189684940-7f3a41a0-21f9-40dc-92c4-afbcf00d2e98.png">

I just ran that hugo mod command from the directory and that was it.  No other steps required.

Marking as the solution.

**Reply by jmichelgarcia** [reply]
Could a note be added to the docs? or even a minor release published?
